Gems/ROS2Sensors/Code/Source/Odometry/ROS2WheelOdometry.cpp

```diff
@@ -19,16 +22,95 @@ namespace ROS2
         const char* WheelOdometryMsgType = "nav_msgs::msg::Odometry";
     }
 
+    // Manual conversion between version without a configuration struct and with a configuration struct.
+    // This is done to maintain backwards compatibility with older versions of the component.
+    // This function checks if in the prefab there exits a member called "Twist covariance" or "Pose covariance".
+    // If it does, it will load the values into the new configuration struct.
+    // If it doesn't, it will treat the loaded json as it would load the new version of the component.
+    // Checking old members is used instead of checking if there is a member called "Odometry configuration"
+    // because is default values are used O3DE does not create an empty member and initializes the component with default values.
+    // Meaning if there does not exist a member called "Odometry configuration" this component could use old values or default ones.
+    AZ::JsonSerializationResult::Result JsonROS2WheelOdometryComponentConfigSerializer::Load(
+        void* outputValue, const AZ::Uuid& outputValueTypeId, const rapidjson::Value& inputValue, AZ::JsonDeserializerContext& context)
+    {
+        namespace JSR = AZ::JsonSerializationResult;
+
+        auto configInstance = reinterpret_cast<ROS2WheelOdometryComponent*>(outputValue);
+        AZ_Assert(configInstance, "Output value for JsonROS2WheelOdometryComponentConfigSerializer can't be null.");
```

Comment:
`reinterpret_cast` is not the same as `dynamic_cast`, so if the expectation here is to validate the pointer type with the assert, all this assert will do is check if `configInstance` is null or not. To be clearer, the assert should be done up front

```suggestion
        AZ_Assert(outputValue, "Output value for JsonROS2WheelOdometryComponentConfigSerializer can't be null.");
        auto configInstance = reinterpret_cast<ROS2WheelOdometryComponent*>(outputValue);
```